pkg/server/image_pull.go

```diff
@@ -273,9 +316,18 @@ func (c *criService) getResolver(ctx context.Context, ref string, cred func(stri
 		logrus.WithError(err).Debugf("Tried registry mirror %q but failed", e)
 		// Continue to try next endpoint
 	}
+
+	if registryTLSConfig, ok := c.config.Registry.TLSConfigs[refspec.Hostname()]; ok {
```

Comment:
Use the `DefaultHost` function to get the host.
https://github.com/containerd/cri/blob/455c66da65d40617df89a2b819970b9cb30a5a77/vendor/github.com/containerd/containerd/remotes/docker/resolver.go#L102

Reply:
Done